src/main/java/com/google/collegeplanner/servlets/CourseListServlet.java

```diff
@@ -47,7 +45,7 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     // Create the URI and specify the parameters.
     String department = request.getParameter("department");
     if (department == null || department == "") {
-      respondWithError("Invalid query parameters.", HttpServletResponse.SC_BAD_REQUEST, response);
+      respondWithError(HttpServletResponse.SC_BAD_REQUEST, response);
```

Comment:
we lost the "reasoning" here in the API response. Can we make a separate function that takes in a custom error message?

Reply:
I agree that a version with a custom message would be useful, but not really here since this is just "bad request"?

Reply:
It does matter here. This is an API, and it should tell the user (our frontend client) what they can do to fix it. Just "bad request" is not enough to narrow down the problem. For this case, "Invalid or missing department" would be a good error message.